realm/realm-core
# Merge master, adapt exceptions

- Exception unification in core/core (#5186)
- Replace specialized exceptions defined in shared_realm.hpp with standard ones (#5433)
- Adapt exceptions thrown in Collections and Results to new exception structure (#5453)
- Adapt AppError to new Exception framework (#5462)
- Add textural representation of category to realm_sync_error_code_t
- Fix Apple build (#5482)
- C API: Make path available through realm_error_t
- Rearrange error codes (#5521)
- Added ErrorCodes::SchemaVersionMismatch
- Assign explicit numbers to error codes
- Fix exception handling after merge
- Throw an exception when trying to aggregate on a collection property
- Fix exception handling after merge
- Fix Windows build
- Update release notes
- Remove exception subclasses which cannot be caught
- Don't swallow access errors when deleting Realms
- Remove a pointless namespace
- Add error_codes.h to the installed headers
- Remove an inaccurate comment claiming that ErrorCodes is generated code
- Avoid building huge map during initialization
- Extract some of the inline definitions from exceptions.hpp
- Store the relevant index in OutOfBoundsException
- Combine the invalid schema change error codes
- Make more exceptions inherit from realm::Exception
- Replace DuplicatePrimaryKeyValue with a more general MigrationFailed
- Adopt realm::Exception in audit
- Fix REQUIRE_THROW_LOGIC_ERROR_WITH_CODE
- Fix CHANGELOG.md
- Fix some formatting errors introduced in merges
- Expose the error category for basic_system_error
- constexprify the collection fixtures
- Make more StringData operations constexpr
- Delete some unused test helper code
- Convert some uses of std exceptions to realm::Exception
- Add missing validation to many of Realm's entry point functions
- Remove the baas dependency for some tests which don't actually talk to the server
- Improve error reporting when initializing ExternalCommitHelper fails
- Improve the error messages in exceptions thrown by collections
- Improve error messages for file operations and opening Realm files
- Validate the message and code for most tests which test exceptions
- Remove unused templates
- Make AppError a RuntimeError
- Delete InvalidEmbeddedOperationException
- Enhance ErrorCodes unittest
- Possible fix for unittest on Mac
- Fix compilation and tests
- Fix warning
- Convert some std exceptions to realm::Exception types
- Fix classification of InvalidQuery
- Add Exception::category() function
- Fix unit tests
- Add test for client reset during async open (#6236)
- Fuzzer Framework (#5940)
- Only use flock emulation for readers-writer locks (#6258)
- Fix websocket redirection after server migration if user is logged in (#6058)
- Prepare release 13.4.0
- Updated release notes
- Separate lifecycle of sync Client from its websocket provider (#6276)
- Fix sync benchmarks (#6281)
- Keep sync event loop running when there are no callbacks/async IO (#6265)
- Fix distro size for fuzzer build variant (#6269)
- Add handling of hex encoded key to realm-trawler (#6264)
- BadChangesetError and TransformError notify users to contact support (#6283)
- Revisit metrics test - update metrics to store whether or not sync to disk was disabled (#6206)
- App 301/308 redirection support doesn't use new location if metadata mode is set to 'NoMetadata' (#6282)
- Notify using the key instead of the index for dictionaries. (#6268)
- Fix wrong handling of WebSocket_Retry_Error (#6290)
- Use a real transaction version for immutable Realms (#6272)
- Add support for nested and overlapping audit scopes
- Ensure that all events generated are identical other than the scope name
- Re-create all objects before updating their properties during client_reset::transfer_group() (#6291)
- [C API] Integration test for websocket provider (#6277)
- Prepare release 13.4.1
- Updated release notes
- OpenSSL 3.0.8 (#6305)
- Improve util/file for Windows (#6221)
- Update yarn download path in install_baas.sh (#6309)
- Update websocket error messages to be more descriptive (#6298)
- Avoid to fail build for fuzzer build variant (#6296)
- Pass callback and user data by valye in realm_mongo_collection_<function> (#6307)
- Wait for sessions to drain during sync client shutdown (#6293)
- prepare core release 13.4.2
- Fix bad changeset loop (#6086)
- Updated release notes
- Support in-memory realms (#6246)
- Fix crash caused by null embedded objects when converting local realm to sync'd realm (#6295)
- Improve RealmTrawler and RealmDump
- Initialize sub-query only once
- subquery microbenchmark
- Make compile
- Fix format
- Fix some unresolved merge conflicts
- Make compile on Windows
- Add document filters to baas service role configuration in tests (#6319)
- Fix ssl stream crash (#6323)
- Make the network code compile (#6262)
- Use proper error message when CreateFile() fails
- Prepare release
- Updated release notes
- remove benchmark compatibility for pre core 6 (#6333)
- Update BindingCallbackThreadObserver to be one per sync client (#6156)
- Bump baas version in evergreen tests (#6337)
- MigrationStore implementation for PBS -> FLX migration (#6324)
- Fix unicode path handling on Windows (#6338)
- Remove Group::compute_aggregated_byte_size() (#6273)
- Fix ApplyToStateCommand for PBS (#6351)
- Fix build due to removed error categories
- Replace std exceptions with realm specific

## What, How & Why?



## ☑️ ToDos
* [ ] 🚦 Tests (or not relevant)
* [ ] C-API, if public C++ API changed.
